pymc/distributions/truncated.py

```diff
@@ -171,22 +174,36 @@ def rv_op(cls, dist, lower, upper, max_n_steps, size=None):
         if size is None:
             size = pt.broadcast_shape(dist, lower, upper)
         dist = change_dist_size(dist, new_size=size)
+        dist_out_idx = dist.owner.outputs.index(dist)
+
+        rv_rng_inputs = [inp for inp in dist.owner.inputs if isinstance(inp.type, RandomType)]
+        rv_non_rng_inputs = [
+            inp for inp in dist.owner.inputs if not isinstance(inp.type, RandomType)
+        ]
+
+        if len(rv_rng_inputs) > 1:
+            raise NotImplementedError("Truncation of variable with multiple RNGs not supported")
```

Comment:
Why? Is it because you wouldn't know which of the multiple rng inputs needs to be truncated? So if you got a mixture distribution, you would not be able to distinguish between the mixing indexes and the mixture components?

Reply:
It's much more mundane than that, multiple RNGs would complicate the logic. The biggest challenge of a Truncated Variable is that in the fallback random case I need to be able to do rejection sampling of a variable in a Scan, for that I need to be able to:
1. Create an equivalent RV inside the Scan
2. Provide the update expressions for any RNG in this RV

As you can see a big part of this PR is just figuring out which position does a RNG input have to go to recreate the base RV. By not supporting cases with multiple RNGs, I only have to worry about one number.

I can try to be less lazy in this PR

Reply:
More importantly, you made me realize Truncated still doesn't work for any SymbolicRandomVariables but only CustomDist ones. I reintroduced the check and added a message to guide users to try and wrap their variables inside a CustomDist.

This is a bit ugly, but the problem is that the graph of stuff like `pm.Mixture` is not all wrapped inside an OpFromGraph, so I don't know what the true "inputs" are to be able to recreate the variable inside the rejection sample scan... Gotta think about this one better.